Aim of this notebook:
-exploring datasets of socio demographic informations

In [55]:
#Importer les packages 
import pandas as pd
from pathlib import Path
import os
import csv


In [36]:
cwd=Path.cwd()
cwd
dir=Path(r"C:\Users\user\Documents\Municipales_2026")
os.chdir(dir)
cwd=Path.cwd()
cwd

WindowsPath('C:/Users/user/Documents/Municipales_2026')

In [37]:
#indiquer le répertoire des données
#donnees="C:/Users/user\Documents\Municipales_2026\Données"
#lire les données
#age 
age=pd.read_csv("Données/Age_csp/agesexcommunes.csv")


C:\Users\user\AppData\Local\Temp\ipykernel_21020\1057842383.py:5: DtypeWarning: Columns (0,2) have mixed types. Specify dtype option on import or set low_memory=False.
  age=pd.read_csv("Données/Age_csp/agesexcommunes.csv")


In [38]:
#visualiser la BDD
age

dep           nomdep codecommune              nomcommune  poph0141962  \
0      25            DOUBS       25443               PALANTINE          5.0   
1      70      HAUTE-SAONE       70440               RECOLOGNE          3.0   
2      21        COTE-D'OR       21276   FONTAINES-EN-DUESMOIS         32.0   
3      31    HAUTE-GARONNE       31266                  LAHAGE          6.0   
4       6  ALPES-MARITIMES        6076                 LIEUCHE          1.0   
...    ..              ...         ...                     ...          ...   
37980  23           CREUSE       23231  SAINT-PIERRE-DE-FURSAC         85.0   
37981  52      HAUTE-MARNE       52238              HARRICOURT          8.0   
37982  48           LOZERE       48049                  CHIRAC         84.0   
37983  61             ORNE       61205                  HEUGON         50.0   
37984  14         CALVADOS       14470        NORREY-EN-BESSIN         32.0   

       poph15391962  poph40591962  poph60p1962  popf0141962  popf15391962  \
0               5.0           6.0          3.0          7.0           6.0   
1               6.0           4.0          2.0          3.0           5.0   
2              37.0          24.0         16.0         29.0          22.0   
3              94.0          17.0         10.0          8.0          14.0   
4               3.0           4.0          2.0          2.0           5.0   
...             ...           ...          ...          ...           ...   
37980         155.0         131.0        124.0        102.0         133.0   
37981          13.0          10.0          8.0          8.0          10.0   
37982         123.0          79.0         68.0        142.0         112.0   
37983          54.0          45.0         33.0         60.0          57.0   
37984          37.0          21.0         11.0         31.0          25.0   

       ...  perprop0141957  perprop60p1957  perpropf1958  perage1958  \
0      ...        0.929899        0.093060      0.946069    0.006167   
1      ...        0.355285        0.711124      0.112636    0.088098   
2      ...        0.938310        0.188820      0.043178    0.144777   
3      ...        0.003273        0.089764      0.001752    0.149059   
4      ...        0.069635        0.323348      0.509821    0.861277   
...    ...             ...             ...           ...         ...   
37980  ...        0.085405        0.952224      0.734750    0.980018   
37981  ...        0.147660        0.659091      0.464339    0.898641   
37982  ...        0.783682        0.893864      0.973157    0.652919   
37983  ...        0.833193        0.591287      0.493604    0.358725   
37984  ...        0.718533        0.267685      0.743903    0.446151   

       perprop0141958  perprop60p1958  perpropf1959  perage1959  \
0            0.929899        0.093060      0.946069    0.006167   
1            0.355285        0.711124      0.112636    0.088098   
2            0.938310        0.188820      0.043178    0.144777   
3            0.003273        0.089764      0.001752    0.149059   
4            0.069635        0.323348      0.509821    0.861277   
...               ...             ...           ...         ...   
37980        0.085405        0.952224      0.734750    0.980018   
37981        0.147660        0.659091      0.464339    0.898641   
37982        0.783682        0.893864      0.973157    0.652919   
37983        0.833193        0.591287      0.493604    0.358725   
37984        0.718533        0.267685      0.743903    0.446151   

       perprop0141959  perprop60p1959  
0            0.929899        0.093060  
1            0.355285        0.711124  
2            0.938310        0.188820  
3            0.003273        0.089764  
4            0.069635        0.323348  
...               ...             ...  
37980        0.085405        0.952224  
37981        0.147660        0.659091  
37982        0.783682        0.893864  
37983        0.833193        0.591287  
37984        0.718533 

In [39]:
list(age.columns)

['dep',
 'nomdep',
 'codecommune',
 'nomcommune',
 'poph0141962',
 'poph15391962',
 'poph40591962',
 'poph60p1962',
 'popf0141962',
 'popf15391962',
 'popf40591962',
 'popf60p1962',
 'ageh1962',
 'agef1962',
 'poph0141968',
 'popf0141968',
 'poph15391968',
 'popf15391968',
 'poph40591968',
 'popf40591968',
 'poph60p1968',
 'popf60p1968',
 'ageh1968',
 'agef1968',
 'poph0141975',
 'popf0141975',
 'poph15391975',
 'popf15391975',
 'poph40591975',
 'popf40591975',
 'poph60p1975',
 'popf60p1975',
 'ageh1975',
 'agef1975',
 'poph0141982',
 'popf0141982',
 'poph15391982',
 'popf15391982',
 'poph40591982',
 'popf40591982',
 'poph60p1982',
 'popf60p1982',
 'ageh1982',
 'agef1982',
 'poph0141990',
 'popf0141990',
 'poph15391990',
 'popf15391990',
 'poph40591990',
 'popf40591990',
 'poph60p1990',
 'popf60p1990',
 'ageh1990',
 'agef1990',
 'poph0141999',
 'popf0141999',
 'poph15391999',
 'popf15391999',
 'poph40591999',
 'popf40591999',
 'poph60p1999',
 'popf60p1999',
 'ageh1999',
 'agef1999',
 '

In [40]:
#regarder ce que sont ces colonnes (age+ date) 
age.ageh2022 #age moyen dans la commune ?? pas sûre 

0        18.000000
1        63.200001
2        44.736843
3        47.094112
4        45.500000
           ...    
37980    18.000000
37981    18.000000
37982    18.000000
37983    18.000000
37984    18.000000
Name: ageh2022, Length: 37985, dtype: float64

noms : 
poph =pop homme
popf= pop femme 

CCL : 
pour chaque année découpage pop H et F pour 0-14, 15-39, 40-59, 60 et + 
age moyen pour F H et Total ( pas sûr)
perage + date ?????? 

37985 communes × 2385 variables

In [41]:
alphabetisation = pd.read_csv("Données/Alphabetisation_csv/alphabetisationcommunes.csv")
alphabetisation

dep codecommune             nomdep               nomcommune  \
0      2B       2B110        HAUTE-CORSE                FAVALELLO   
1      2B       2B179        HAUTE-CORSE                   NOVALE   
2      2B       2B214        HAUTE-CORSE                    PIANO   
3      2B       2B005        HAUTE-CORSE                   ALANDO   
4      2B       2B151        HAUTE-CORSE              LUNGHIGNANO   
...    ..         ...                ...                      ...   
37671  17       17113  CHARENTE-MARITIME                   CLOTTE   
37672  25       25167              DOUBS                  COTTIER   
37673  52       52411        HAUTE-MARNE            PUELLEMONTIER   
37674  82       82155    TARN-ET-GARONNE  SAINT-ANTONIN-NOBLE-VAL   
37675  67       67198           BAS-RHIN                HINSBOURG   

       conjsign1686  conjnosi1686  conjsign1786  conjnosi1786  conjsign1816  \
0          0.010000      0.370000      0.040000      0.570000      0.060000   
1          0.050000      1.410000      0.140000      2.200000      0.220000   
2          0.020000      0.490000      0.050000      0.770000      0.080000   
3          0.020000      0.480000      0.050000      0.750000      0.070000   
4          0.010000      0.170000      0.020000      0.260000      0.030000   
...             ...           ...           ...           ...           ...   
37671      4.890000      4.590000      9.080000      3.820000      8.880000   
37672      0.000000      0.070000      0.020000      0.080000      0.030000   
37673     32.880001      6.920000     76.519997      0.000000    103.660000   
37674      5.390000     37.470001      9.740000     46.639999     16.209999   
37675      1.130000      0.000000      1.790000      0.000000      2.150000   

       conjnosi1816  ...  palpha1942  peralpha1942  palpha1943  peralpha1943  \
0          0.710000  ...    0.315789  6.019342e-04    0.315789  5.857286e-04   
1          2.690000  ...    0.294118  1.470252e-04    0.303030  5.705081e-04   
2          0.950000  ...    0.300000  5.764499e-04    0.300000  4.736071e-04   
3          0.920000  ...    0.300000  5.805184e-04    0.300000  4.795179e-04   
4          0.320000  ...    0.272727  4.068623e-07    0.272727  4.063711e-07   
...             ...  ...         ...           ...         ...           ...   
37671      5.320000  ...    1.000000  8.045195e-01    1.000000  5.825751e-01   
37672      0.070000  ...    1.000000  3.924490e-01    1.000000  8.636994e-01   
37673      0.000000  ...    1.000000  9.981855e-01    1.000000  5.778921e-01   
37674     45.860001  ...    1.000000  9.449841e-01    1.000000  5.296085e-01   
37675      0.000000  ...    1.000000  8.731292e-01    1.000000  4.726169e-01   

       palpha1944  peralpha1944  palpha1945  peralpha1945  palpha1946  \
0        0.277778  1.070061e-06    0.294118  6.080951e-06    0.294118   
1        0.296875  3.564410e-05    0.306452  5.184287e-04    0.300000   
2        0.300000  2.190673e-04    0.300000  3.386905e-05    0.300000   
3        0.300000  2.017249e-04    0.300000  4.861075e-05    0.300000   
4        0.272727  4.058852e-07    0.272727  4.053967e-07    0.300000   
...           ...           ...         ...           ...         ...   
37671    1.000000  4.446173e-01    1.000000  8.109778e-01    1.000000   
37672    1.000000  4.583446e-01    1.000000  3.493365e-01    1.000000   
37673    1.000000  8.688728e-01    1.000000  8.188458e-01    1.000000   
37674    1.000000  4.193314e-01    1.000000  9.371914e-01    1.000000   
37675    1.000000  9.320233e-01    1.000000  1.000001e+00    1.000000   

       peralpha1946  
0      6.257733e-07  
1      2.834385e-06  
2      3.938690e-06  
3      5.042997e-06  
4      5.411098e-06  
...             ...  
37671  9.998457e-01  
37672  9.998457e-01  
37673  9.999402e-01  
37674  9.999985e-01  
37675  1.000002e+00  

[37676 rows x 388 columns]

In [42]:
list(alphabetisation.columns)

['dep',
 'codecommune',
 'nomdep',
 'nomcommune',
 'conjsign1686',
 'conjnosi1686',
 'conjsign1786',
 'conjnosi1786',
 'conjsign1816',
 'conjnosi1816',
 'conjsign1854',
 'conjnosi1854',
 'conjsign1856',
 'conjnosi1856',
 'conjsign1861',
 'conjnosi1861',
 'conjsign1866',
 'conjnosi1866',
 'conjsign1871',
 'conjnosi1871',
 'conjsign1876',
 'conjnosi1876',
 'conjsign1882',
 'conjnosi1882',
 'conjsign1887',
 'conjnosi1887',
 'conjsign1890',
 'conjnosi1890',
 'conjsign1895',
 'conjnosi1895',
 'conjsign1900',
 'conjnosi1900',
 'conjsign1905',
 'conjnosi1905',
 'alpha1866',
 'nonal1866',
 'alpha1867',
 'nonal1867',
 'alpha1868',
 'nonal1868',
 'alpha1869',
 'nonal1869',
 'alpha1870',
 'nonal1870',
 'alpha1871',
 'nonal1871',
 'alpha1872',
 'nonal1872',
 'alpha1873',
 'nonal1873',
 'alpha1874',
 'nonal1874',
 'alpha1875',
 'nonal1875',
 'alpha1876',
 'nonal1876',
 'alpha1877',
 'nonal1877',
 'alpha1878',
 'nonal1878',
 'alpha1879',
 'nonal1879',
 'alpha1880',
 'nonal1880',
 'alpha1881',
 'nona

In [43]:
glossaire = {
    
    "conjsign" : "conjoints signant acte de mariage",
    "conjnosi" : "Conjoints ne signant pas ???", 
    "alpha" : "alphabetisation ?",
    

}


In [71]:
#ouvrir un fichier dta pour obtenir les labels

itr = pd.read_stata('C:/Users/user/Downloads/Alphabetisation_dta/alphabetisationcommunes.dta', iterator=True)
glossaire_alphabetisation=itr.variable_labels()
#type(glossaire_alphabetisation)
glossaire_alphabetisation

{'dep': 'Code du departement',
 'codecommune': 'Code national de la commune',
 'nomdep': 'Nom du departement',
 'nomcommune': 'Nom de la commune',
 'conjsign1686': 'Conjoints (hommes et femmes maries) signant leur acte de mariage',
 'conjnosi1686': 'Conjoints (hommes et femmes maries) ne signant pas leur acte de mariage',
 'conjsign1786': 'Conjoints (hommes et femmes maries) signant leur acte de mariage',
 'conjnosi1786': 'Conjoints (hommes et femmes maries) ne signant pas leur acte de mariage',
 'conjsign1816': 'Conjoints (hommes et femmes maries) signant leur acte de mariage',
 'conjnosi1816': 'Conjoints (hommes et femmes maries) ne signant pas leur acte de mariage',
 'conjsign1854': 'Conjoints (hommes et femmes maries) signant leur acte de mariage',
 'conjnosi1854': 'Conjoints (hommes et femmes maries) ne signant pas leur acte de mariage',
 'conjsign1856': 'Conjoints (hommes et femmes maries) signant leur acte de mariage',
 'conjnosi1856': 'Conjoints (hommes et femmes maries) ne sig

In [69]:
#with open('glossaire_alphabetisation.csv', 'w') as f:  
#        w = csv.DictWriter(f, glossaire_alphabetisation.keys())
#        w.writeheader()        
#        w.writerows(glossaire_alphabetisation) 

import csv

with open("glossaire_alphabetisation.csv", "w", newline="", encoding="utf-8") as f:
    writer = csv.writer(f)
    
    # en-tête
    writer.writerow(["terme", "definition"])
    
    # une ligne par entrée du dictionnaire
    for terme, definition in glossaire_alphabetisation.items():
        writer.writerow([terme, definition])

